File: 24_10/17/P42885/solution.js
function solution(people, limit) {
    // 사람 몸무게 정렬     
    people.sort((a,b) => a - b);
    // 가벼운쪽, 무거운쪽 인덱스
    let left = 0, right = people.length - 1;
    
    // 무거운쪽부터 태우고 남은 무게가 있으면 가벼운쪽 태우기
    let count = 0
    while (left <= right) {
        const remain = limit - people[right--];
        if (people[left] <= remain) left++
        count++
    }

    return count;
}

File: 24_10/17/P42885/Solution.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        // 몸무게 정렬
        Arrays.sort(people);
        // 현재까지 구출한 사람 인덱스(가벼운 사람부터, 무거운 사람부터)
        int left = 0, right = people.length - 1;
        
        // 보트 개수 구하기
        int count = 0;
        while (left <= right) {
            // 무거운 사람은 일단 태운다고 가정
            int remain = limit - people[right--];
            // 남은 무게를 보고 왼쪽 사람 태우기
            if (people[left] <= remain) left++;
            // 보트 개수 증가    
            count++;
        }

        return count;
    }
}